cip-0033.md

```diff
@@ -0,0 +1,87 @@
+    Status: Draft
+    Type: Enhancement (Hard Fork)
+    Created: 2021-12-11
+    
+# Introduction
+Most platforms use smart contracts to handle token creation and serve various activities like buying, selling, and transferring from within the smart contract. 
+Signum’s approach is different. On the Signum blockchain, tokens are situated similarly to the blockchain’s own native coin (Signa) in that they are unique entities that can be transferred or traded directly between accounts.
+Their position is directly attached to accounts, not taken via a smart contract. 
+
+All possible activities are governed by the blockchain nodes’ existing transaction types and logic, making the current token setup “smart”, even though we need to add additional transactions and logic to refine it for a true “smart token” setup.
+
+# Motivation
+In 2014, Signum introduced its unique token functionality with the genesis block.  Since that time, not much development has taken place in this particular area of the blockchain.  Surprisingly, when we compare the current features of ERC-20/BEP-20 contracts with the existing Signum setup, we see that our original setup was already forward-looking.
+
+![Compare](cip-0033/CIP33_table.png)
+
+
+The goal of CIP-33 is to close gaps for our Signum token by adding base functionality and new transaction types to fulfill requirements for mass adoption. Also, to further prepare for the new on-chain native liquidity pools.
+
+To express the new scope with a new term, the new tokens will be named  **Smart Token** after the needed hard-fork.
+
+## Basic enhancements for Smart Tokens
+
+### Creating smart tokens
+When creating a smart token, the user will be able to define the token as mintable or non-mintable.  If the token is defined as mintable, the creator will be able to mint additional tokens.  If non-mintable, the token is limited to the initial supply specified at the time of creation.  All existing tokens created prior to the upcoming hard-fork are non-mintable.
+
+The new transaction  for minting tokens will incur a minimum network transaction fee.
+(MIN-FEE)
+
+### Burning smart tokens
+Any Signum token (including those that existed prior to the hard fork) can be burned by transferring it to address 0 (zero) using the transfer transaction.  After the hard fork, it will be possible to burn Signa with as well by sending to the address 0 (zero).
+
+## Special extensions for Smart Tokens
+
+### Define TreasuryAccounts
+A new transaction type (**addAssetTreasuryAccount**) will be created that allows a token creator to define Treasury Accounts for a given smart token. Once set, this transaction will not be able to be reverted.  Token balances in these accounts will not be used when calculating circulating supply, nor will the balance in burn address 0.
+
+With this new transaction, the smart token creator can publicly specify on-chain which accounts belong to a use-case.  It will be possible to do this for existing tokens as well.
```

Comment:
```suggestion
With this new transaction, the smart token creator can publicly specify on-chain which accounts belong to a use-case. It will be possible to do this for existing tokens as well.
```

Reply:
done